include/verilated_trace_threaded.h

```diff
@@ -0,0 +1,103 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//=============================================================================
+//
+// THIS MODULE IS PUBLICLY LICENSED
+//
+// Copyright 2001-2020 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//=============================================================================
+///
+/// \file
+/// \brief Threaded tracing internals
+///
+//=============================================================================
+// SPDIFF_OFF
+
+#ifndef _VERILATED_TRACE_THREADED_H_
+#define _VERILATED_TRACE_THREADED_H_ 1
+
+#include <deque>
+#include <condition_variable>
+#include <mutex>
+
+// A simple synchronized first in first out queue
+template <class T> class VerilatedThreadQueue {
+private:
+    std::deque<T> m_queue;
+    std::condition_variable m_cv;
```

Comment:
```suggestion
    std::condition_variable m_cv VL_GUARDED_BY(m_mutex);
```

And be sure to try compiling with clang; it will make sure you get m_mutex in right places.